src/auth/discord.rs

```diff
@@ -1 +1,404 @@
+    match discord_auth {
+        Ok(access_token) => {
+            dbg!(&access_token);
+            let (user_name, user_id) = apis::user_details(&access_token).await?;
+            let user_detail_obj: UserDetail = UserDetail {
+                token: access_token.to_owned(),
+                user_name,
+                user_id,
+            };
+            let user_detail_str = serde_json::to_string(&user_detail_obj)?;
+            let secret_key = fpm::auth::secret_key();
+            let mc_obj = magic_crypt::new_magic_crypt!(secret_key.as_str(), 256);
+            return Ok(actix_web::HttpResponse::Found()
+                .cookie(
+                    actix_web::cookie::Cookie::build(
+                        fpm::auth::AuthProviders::Discord.as_str(),
+                        mc_obj
+                            .encrypt_to_base64(&user_detail_str)
+                            .as_str()
+                            .to_owned(),
+                    )
+                    .domain(fpm::auth::utils::domain(req.connection_info().host()))
+                    .path("/")
+                    .permanent()
+                    .finish(),
+                )
+                .append_header((actix_web::http::header::LOCATION, "/".to_string()))
+                .finish());
+        }
+        Err(err) => Ok(actix_web::HttpResponse::InternalServerError().body(err.to_string())),
+    }
+}
+// it returns identities which matches to given input
+pub async fn matched_identities(
+    ud: UserDetail,
+    identities: &[fpm::user_group::UserIdentity],
+) -> fpm::Result<Vec<fpm::user_group::UserIdentity>> {
+    let discord_identities = identities
+        .iter()
+        .filter(|identity| identity.key.starts_with("discord"))
+        .collect::<Vec<&fpm::user_group::UserIdentity>>();
+
+    if discord_identities.is_empty() {
+        return Ok(vec![]);
+    }
+
+    let mut matched_identities = vec![];
+    // matched_user_servers
+    matched_identities.extend(matched_user_servers(&ud, discord_identities.as_slice()).await?);
+    // matched_thread_members
+    matched_identities.extend(matched_thread_members(&ud, discord_identities.as_slice()).await?);
+    //matched_event_members
+    matched_identities.extend(matched_event_members(&ud, discord_identities.as_slice()).await?);
+    Ok(matched_identities)
+}
+pub async fn matched_user_servers(
+    ud: &UserDetail,
+    identities: &[&fpm::user_group::UserIdentity],
+) -> fpm::Result<Vec<fpm::user_group::UserIdentity>> {
+    use itertools::Itertools;
+
+    let user_servers = identities
+        .iter()
+        .filter_map(|i| {
+            if i.key.eq("discord-server") {
+                Some(i.value.as_str())
+            } else {
+                None
+            }
+        })
+        .collect_vec();
+
+    if user_servers.is_empty() {
+        return Ok(vec![]);
+    }
+    let user_joined_servers = apis::user_servers(ud.token.as_str()).await?;
```

Comment:
Check if this API is paginated, if yes then put todo on it. We will implement it later.

Reply:
its already mentioned